src/libAtomVM/term.h

```diff
@@ -1080,4 +1080,33 @@ static inline int term_is_number(term t)
  */
 void term_display(FILE *fd, term t, const Context *ctx);
 
+/**
+ * @brief Checks if a term is a string (i.e., a list of characters)
+ *
+ * @details Returns 1 if a term is a list of characters or an empty list (nil term), otherwise 0.
+ * @param t the term that will be checked.
+ * @return 1 if check succedes, 0 otherwise.
+ */
+static inline int term_is_string(term t)
+{
+    if (term_is_list(t)) {
+        int proper;
+        term_list_length(t, &proper);
```

Comment:
I don't think this a good solution, we are iterating 2 times over the same list.
The first time here we iterate on it but then we discard the list len.

Reply:
Fixed to iterate only once.